src/polycubed/src/CubesDump.cpp

```diff
@@ -0,0 +1,293 @@
+                                 const nlohmann::json& body,
+                                 const ListKeyValues &keys,
+                                 polycube::polycubed::Rest::Resources::Endpoint::Operation opType,
+                                 polycube::polycubed::Rest::Resources::Endpoint::ResourceType resType) {
+
+  if (configuration::config.getCubesNoDump()) {
+    return;
+  }
+
+  std::string resourcePath = resource.substr(polycube::polycubed::RestServer::base.size(), resource.size());
+
+  std::stringstream ssResource(resourcePath);
+  std::vector<std::string> resItem;
+  std::string tokenResource;
+
+  while(std::getline(ssResource, tokenResource, '/')) { /* Get path elements */
+    resItem.push_back(tokenResource);
+  }
+
+  // Depending on the operation, we look into the resource string, the body and the ListKeyValues
+  // to update the configuration of the cubes in the map (<cubeName, cubeConfiguration>) we have in memory.
+  // If it is not the initial topology load, the whole configuration is saved to file
+  // after each modification by a separate thread, to avoid to overload the server thread.
+  std::lock_guard<std::mutex> guardConfigMutex(cubesConfigMutex);
+  switch (opType) {
+    case polycube::polycubed::Rest::Resources::Endpoint::Operation::kCreate:
+    case polycube::polycubed::Rest::Resources::Endpoint::Operation::kReplace: {
+      UpdateCubesConfigCreateReplace(resItem, body, keys, resType);
+      break;
+    }
+
+    case polycube::polycubed::Rest::Resources::Endpoint::Operation::kUpdate: {
+      UpdateCubesConfigUpdate(resItem, body, keys, resType);
+      break;
+    }
+
+    case polycube::polycubed::Rest::Resources::Endpoint::Operation::kDelete: {
+      UpdateCubesConfigDelete(resItem, body, keys, resType);
+      break;
+    }
+  }
+
+
+  // if startup is true, then it is the first load and the save thread must not be notified
+  if (!polycube::polycubed::RestServer::isStartup()) {
+    toSave++;
+    waitForUpdate.notify_one();
+  }
+}
+
+void CubesDump::UpdateCubesConfigCreateReplace(const std::vector<std::string> &resItem,
+                                               const nlohmann::json& body,
+                                               const ListKeyValues &keys,
+                                               polycube::polycubed::Rest::Resources::Endpoint::ResourceType resType) {
+
+  const std::string &serviceName = resItem[0];
+  const std::string &cubeName = resItem[1];
+
+  // map saving key values corresponding to proper list name
+  std::map<std::string, std::vector<polycube::polycubed::Rest::Resources::Body::ListKeyValue>> keyValues;
+  for (auto &key : keys) {
+    keyValues[key.list_element].push_back(key);
+  }
+
+  switch (resType) {
+    case polycube::polycubed::Rest::Resources::Endpoint::ResourceType::Service: {
+      nlohmann::json serviceField = nlohmann::json::object();
+      serviceField["service-name"] = serviceName;
```

Comment:
Is this really needed?, Isn't `service-name` already present in the json object?